Library/Homebrew/software_spec.rb

```diff
@@ -91,15 +91,17 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner = self
-      r.version ||= begin
+      next if r.version
+
+      r.version(begin
         raise "#{full_name}: version missing for \"#{r.name}\" resource!" if version.nil?
 
         if version.head?
           Version.create("HEAD")
         else
           version.dup
         end
-      end
+      end)
```

Comment:
Maybe something like this would make more sense while you're in this file.

```rb
raise "#{full_name}: version missing for \"#{r.name}\" resource!" if version.nil?

r.version(version.head? ? Version.create("HEAD") : version.dup)
```